lib/src/response.dart

```diff
@@ -2,94 +2,49 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import 'dart:async';
 import 'dart:convert';
-import 'dart:typed_data';
 
-import 'package:http_parser/http_parser.dart';
-
-import 'base_request.dart';
-import 'base_response.dart';
-import 'streamed_response.dart';
+import 'message.dart';
 import 'utils.dart';
 
 /// An HTTP response where the entire response body is known in advance.
-class Response extends BaseResponse {
-  /// The bytes comprising the body of this response.
-  final Uint8List bodyBytes;
-
-  /// The body of the response as a string. This is converted from [bodyBytes]
-  /// using the `charset` parameter of the `Content-Type` header field, if
-  /// available. If it's unavailable or if the encoding name is unknown,
-  /// [LATIN1] is used by default, as per [RFC 2616][].
+class Response extends Message {
```

Comment:
At some point this should utility getters like `shelf.Response.expires` *et al*. That doesn't need to happen in this PR though.